### PR TITLE
python3Packages.tempest: remove unittest2 requirement

### PR DESCRIPTION
**Pull request by jonringer**
###### Motivation for this change

related: https://github.com/NixOS/nixpkgs/pull/154650
related: https://github.com/NixOS/nixpkgs/pull/155005

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/tempest/default.nix [hunk 1]
```diff
@@ -36,6 +36,14 @@ buildPythonApplication rec {
     sha256 = "0521d3042360c0fb469b16f99174a9abddbae8a2d2a81268cfc664f1ccfdd0f9";
   };
 
+  patches = [
+    # remove need for unittest2
+    (fetchpatch {
+      url = "https://github.com/openstack/tempest/pull/32/commits/cd3745c27b7d8fcdaffc72b965a3d803d9ee12c2.patch";
```
Comment (dotlambda) [inline pkgs/development/python-modules/tempest/default.nix:42]:
```suggestion
      url = "https://github.com/openstack/tempest/commits/cd3745c27b7d8fcdaffc72b965a3d803d9ee12c2.patch";
```

Comment (SuperSandro2000) [reply pkgs/development/python-modules/tempest/default.nix:42]: Openstack does not work with pull requests. You need to go through their lovely Gerrit and unless you know someone no one probably looks at the changes.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 155057` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
2 packages built:
</summary>
<ul>
<li>python310Packages.tempest</li>
<li>python39Packages.tempest</li>
</ul>
</details>

**Comment by dotlambda** [comment]
@SuperSandro2000 Why did you just merge even though I suggested a change?

**Comment by SuperSandro2000** [comment]
Because I was lazy and I stopped careing about is only cosmetically and people get way to often way to angry over such changes.

**Comment by dotlambda** [comment]
It's not a cosmetical change though, PR commit URLs throw a 404 if the commit is removed by a force-push.

**Comment by SuperSandro2000** [comment]
Really? I didn't know that.

**Comment by dotlambda** [comment]
I lied. It's a 406 :D

**Comment by SuperSandro2000** [comment]
https://github.com/NixOS/nixpkgs/pull/155279

File: pkgs/development/python-modules/tempest/default.nix (hunk 1 above)
Comment (jonringer) [reply pkgs/development/python-modules/tempest/default.nix:42]: this is annoying, why don't they say it's a read-only mirror then?

**Pull request merged**